src/com/cloud/containercluster/ContainerClusterManagerImpl.java

```diff
@@ -1966,20 +1968,26 @@ public KeyFactory getKeyFactory() {
     }
 
     public X509Certificate pemToX509Cert(final String pem) throws IOException {
-        final PEMReader pr = new PEMReader(new StringReader(pem));
-        return (X509Certificate) pr.readObject();
+        return (X509Certificate) certService.parseCertificate(pem);
     }
 
-    public String x509CertificateToPem(final X509Certificate cert) throws IOException {
-        final StringWriter sw = new StringWriter();
-        try (final PEMWriter pw = new PEMWriter(sw)) {
-            pw.writeObject(cert);
+    /**
+     * @deprecated this should move to {@link CertService}
+     */
+    @Deprecated
+    public String x509CertificateToPem(final X509Certificate cert) throws IOException, CertificateEncodingException {
+
+        // TODO convert cert to PemObject
```

Comment:
what exactly is meant by this TODO? is it dependant on the other PR?

Reply:
it means I should write better TODOs. It seems it is already converted to a PemObject below :(